src/Stack/Setup.hs

```diff
@@ -1085,14 +1091,27 @@ installGHCPosix version downloadInfo _ archiveFile archiveType tempDir destDir =
 
     let runStep step wd env cmd args = do
             menv' <- modifyEnvOverride menv (Map.union env)
-            result <- withWorkingDir wd
-                    $ withEnvOverride menv'
-                    $ withProc cmd args
-                    $ try
-                    . readProcessStdout_
-                    -- Calling the ./configure script requires that stdin is
-                    -- open
-                    . setStdin (useHandleOpen stdin)
+            result <- case logLevel of
+              LevelDebug -> do
+                let logLines = CB.lines .| CL.mapM_ (logInfo . T.decodeUtf8With T.lenientDecode)
```

Comment:
It's slightly less efficient, but assuming it gives the output you're looking for, you could use `logDebug` here instead, and then unconditionally use this implementation, and totally bypass analyzing the `logLevel`. I'm not too concerned about the performance overhead of breaking the output from `./configure` into lines and then potentially ignoring them, so I'd lean in that direction for the code simplification it would yield.

Reply:
I am not sure I follow how we can avoid analyzing `loglevel`. If `logLevel` is `LogDebug` (i.e, stack-setup was invoked with `-v`) we need to read stdout of configure script and print it in `stack` command's stdout. Otherwise, we should not print anything. If we used `logDebug` in both the cases, wouldn't users  who ran stack-setup without `-v` complain about the verbosity?

Reply:
`logDebug` will only produce output if Stack is run with `-v`.